src/samv71_serial_ccsds/samv71_serial_ccsds.c

```diff
@@ -277,17 +320,32 @@ void SamV71SerialCcsdsPoll(void *private_data) {
 
   int errorCode = 0;
   while (true) {
-    Uart_read(&self->m_hal_uart.uart, self->m_recv_buffer, UINT32_MAX,
-              &errorCode);
-    if (errorCode == Uart_ErrorCodes_Timeout) {
+    /// Wait for data to arrive. Semaphore will be given
+    xSemaphoreTake(self->m_rx_semaphore, portMAX_DELAY);
+
+    size_t recvBytesCount = 0;
+    for (recvBytesCount = 0;
+         (recvBytesCount < Serial_CCSDS_SAMV71_RECV_BUFFER_SIZE) |
+         (recvBytesCount < Serial_CCSDS_SAMV71_FIFO_BUFFER_SIZE);
```

Comment:
Why second comparison is necessary?

Reply:
I realize it is not neccessary. It was deleted in new commit.